src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBaseScrollContentPresenter.Android.cs

```diff
@@ -10,10 +10,15 @@
 
 namespace Windows.UI.Xaml.Controls
 {
-	public sealed partial class ListViewBaseScrollContentPresenter : ContentPresenter, IScrollContentPresenter
+	public sealed partial class ListViewBaseScrollContentPresenter : IScrollContentPresenter, INativeScrollContentPresenter
 	{
-
-		private Thickness _oldPadding;
+		// NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE NOTE
+		//
+		// This class is about to be replaced by an implementation of INativeScrollContentPresenter which only adapts the 'NativePanel'.
```

Comment:
I'm curious what you have in mind here? It would be great to be able to eliminate the need for a custom ScrollViewer template for ListView and GridView, and instead enable the correct behaviour transparently.

Reply:
Well it's almost already the case as I made the `ListViewSCP` inherit from `SCP` and stripped of  lot of logic which was shared with the `NativeSCP` (moved directly in the SCP).

The ides is that the `SCP` on iOS and Android should only act as a decorator/proxy of my new `INativeSCP`. Then we should have 2 implementations of the `INativeSCP`: one for the default `ScrollViewer` (which _could_ inherit from the `UIScrollView`) and one for the `ListView` (which would adapt the `NativePanel` managed by the `ListView`). There is then no difficulty to instance one of those dynamically depending of the templated parent when we create the visual tree. Actually we are already doing that for the default `ScrollViewer`, we can do the same for the `ListView`.

The main difficulty is to re-dispatch responsibilities. We are dealing with old code here, and I want to make sure that the `INativeSCP` is as minimal as possible.